bundles/org.openhab.binding.opensmartcity/src/main/java/org/openhab/binding/opensmartcity/internal/OpenSmartCityWeatherHandler.java

```diff
@@ -137,4 +182,68 @@ public void dispose() {
 
     public class JsonResponse {
     }
+
+    private String apiRequest(String queryPath) {
+        try {
+            String url = bridgeHandler.basePath + queryPath;
+            logger.debug("Requesting {}", url);
+
+            Request request = bridgeHandler.httpClient.newRequest(url);
+            // TODO: remove hard coded credentials
+            request.getHeaders().add("Authorization",
+                    "Basic " + Base64.getEncoder().encodeToString(("smarthomeuser:Solingen2030!").getBytes()));
+            ContentResponse response = request.send();
+
+            if (response.getStatus() == 200) {
+                String content = response.getContentAsString();
+                logger.debug("Requesting URL {}", url);
+                logger.debug("Retreived Content {}", content);
+                return content;
+            }
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+        return "";
+    }
+
+    private void getNearestOnlineLocation() {
```

Comment:
I expect a return value from a get-method. Maybe rather rename it to `updateNearestOnlineLocation`?

Reply:
Changed in next commit